### DISCUSSION TITLE
Trouble Retrieving Directory Size and Quota Using go-ceph/cephfs

### DISCUSSION THREAD

**Discussion by ThingOfNull** [ceph/go-ceph#948] [Q&A] [answered]
Hello everyone, I am trying to use go-ceph/cephfs to get the size of a specific directory on cephfs and the quota set for it. I tried using entry.Statx(), but the Size I got is 0, and Blocks is also 0. Here's my code:

```
package main

import (
	"fmt"

	"github.com/ceph/go-ceph/cephfs"
)

func main() {

	confFile := "/home/dev/xxxxx/xxxxxxxx/xxxxxxxxx/ceph.conf"
	mount, err := cephfs.CreateMount()
	if err != nil {
		fmt.Printf("Unable to create mount: %s", err)
		return
	}

	if err := mount.ReadConfigFile(confFile); err != nil {
		fmt.Printf("Unable to read config file: %s", err)
		return
	}

	if err := mount.SetConfigOption("client_mountpoint", "/"); err != nil {
		fmt.Printf("Unable to set client_mountpoint: %s", err)
		return
	}
	if err := mount.Init(); err != nil {
		fmt.Printf("Unable to init mount: %s", err)
		return
	}
	if err := mount.Mount(); err != nil {
		fmt.Printf("Unable to mount CephFS: %s", err)
		return
	}
	dir, err := mount.OpenDir("/")
	if err != nil {
		fmt.Printf("Unable to open root directory: %s", err)
		return
	}
	defer dir.Close()

	for {
		entry, err := dir.ReadDirPlus(0, 0)
		if err != nil {
			fmt.Printf("Unable to read root directory: %s", err)
			return
		}
		if entry == nil {
			break
		}

		fmt.Println(entry.Name())
		statx := entry.Statx()
		if statx != nil {
			fmt.Printf("Mode: %v, Size: %v, Blocks: %v\n", statx.Mode, statx.Size, statx.Blocks)
			fmt.Printf("%v", statx)
		}
	}
	defer mount.Unmount()
}

```

Can anyone help me understand what might be going wrong?

**Comment by ThingOfNull** [comment]
In the past, I obtained the directory size and quota by mounting ceph-fuse to the device and then using the commands 
```getfattr -d -m ".*" /mnt/cephfs/xxxxxx ```
and 
```getfattr -d -n ceph.quota  /mnt/cephfs/xxxxxx ```
to get values like ceph.dir.rbytes, ceph.dir.rentries, ceph.quota, etc. However, I don't think this is an elegant or optimal method, so I want to use go-ceph to get this kind of information.

**Comment by nixpanic** [comment, marked as answer]
The `dir.ReadDirPlus(0, 0)` that you do, passes `0` for `StatxMask`. You will need to pass a mask of values like `StatxMode` as defined on https://pkg.go.dev/github.com/ceph/go-ceph@v0.25.0/cephfs#pkg-constants.

Alternatively you could use the extended attribute method with https://pkg.go.dev/github.com/ceph/go-ceph@v0.25.0/cephfs#File.GetXattr

**Reply by ThingOfNull** [reply]
Yes, you're right, I should use the GetXattr method.
thanks
